pkg/storegateway/batch_series_test.go

```diff
@@ -771,52 +773,169 @@ func TestBlockSeriesChunkRefsSetsIterator_ErrorPropagation(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I would rename the test to `TestOpenBlockSeriesChunkRefsSetsIterator`. At least I would remove the `_ErrorPropagation` suffix given now it's not just testing error propagation.